test_conformance/extensions/cl_khr_command_buffer/command_buffer_profiling.cpp

```diff
@@ -0,0 +1,289 @@
+        {
+            // enqueue simultaneous command-buffers with profiling command queue
+            error = RunSimultaneous();
+            test_error(error, "RunSimultaneous failed");
+        }
+        else
+        {
+            // enqueue single command-buffer with profiling command queue
+            error = RunSingle();
+            test_error(error, "RunSingle failed");
+        }
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define ADD_PROF_PARAM(prop)                                                   \
+    {                                                                          \
+        prop, #prop, 0                                                         \
+    }
+    struct ProfilingParam
+    {
+        cl_profiling_info param;
+        std::string name;
+        cl_ulong value;
+    };
+
+    //--------------------------------------------------------------------------
+    cl_int VerifyResult(const clEventWrapper& event)
+    {
+        cl_int error = CL_SUCCESS;
+        cl_int status;
+        error = clGetEventInfo(event, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                               sizeof(status), &status, NULL);
+        test_error(error, "clGetEventInfo() failed");
+
+        if (status != CL_SUCCESS)
+            test_fail("Kernel execution status %d! (%s:%d)\n", status, __FILE__,
+                      __LINE__);
+
+        std::vector<ProfilingParam> prof_params = {
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_QUEUED),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_SUBMIT),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_START),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_END),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_COMPLETE)
+        };
+
+        // gather profiling timestamps
+        for (auto&& p : prof_params)
+        {
+            error = clGetEventProfilingInfo(event, p.param, sizeof(p.value),
+                                            &p.value, NULL);
+            test_error(error, "clGetEventProfilingInfo() failed");
+        }
+
+        // verify the results by comparing timestamps
+        for (int i = 1; i < prof_params.size(); i++)
+        {
+            if (prof_params[i - 1].value > prof_params[i].value)
+            {
+                log_error(
+                    "Profiling %s should be smaller than or equal to %s for "
```

Comment:
Printing the values failed on as well as the names would make this message even more informative

Reply:
Done